Add detection NN example

szabi-luxonis opened the pull request:
Added getLayerInt32 for NN results, used by deeplabv3 for example
Added mobilenet device side decoding example
Added tiny yolo v3 device side decoding example
Improved Detection network API

Fix FW bugs: https://gitlab.com/luxonis/depthai/-/merge_requests/104

themarpe commented on include/depthai/pipeline/node/DetectionNetwork.hpp:
```diff
@@ -26,6 +26,7 @@ class DetectionNetwork : public NeuralNetwork {
    public:
     Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::ImgFrame, true}}};
     Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::Buffer, false}}};
+    Output passthrough{*this, "passthrough", Output::Type::MSender, {{DatatypeEnum::Buffer, true}}};
```
`DatatypeEnum::ImgFrame, true` same as input (if it really does passthrough input only)

themarpe commented on examples/src/camera_mobilenet_example.cpp:
```diff
@@ -30,7 +31,11 @@ dai::Pipeline createNNPipeline(std::string nnPath){
     // Link plugins CAM -> NN -> XLINK
     colorCam->preview.link(nn1->input);
-    colorCam->preview.link(xlinkOut->input);
+    if (syncNN)
```
Inline or `{}`

themarpe commented on include/depthai/pipeline/datatype/ImgDetections.hpp:
```diff
@@ -18,8 +18,7 @@ class ImgDetections : public Buffer {
 
-    // reference
-    std::vector<ImgDetection>& detections;
+    std::vector<ImgDetection> getDetections(void) const;
```
```
std::vector<ImgDetection> getDetections() const;
```

themarpe reviewed (changes requested):
LGTM apart from the requested changes.
Rationally behind the ImgDetections move from reference to getter?
And in case it remains, add a setter as well to be able to create ImgDetections message and send it to the device

themarpe commented on include/depthai/pipeline/node/DetectionNetwork.hpp:
```diff
@@ -26,6 +26,7 @@ class DetectionNetwork : public NeuralNetwork {
    public:
     Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::ImgFrame, true}}};
     Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::Buffer, false}}};
```
This should be `DatatypeEnum::ImgDetections`

themarpe commented on include/depthai/pipeline/Pipeline.hpp:
```diff
@@ -147,6 +147,10 @@ class Pipeline {
         return impl()->assetManager;
     }
+
+    void setPipelineOpenVINOVersion(OpenVINO::Version version) {
```
I'd rather go with `setOpenVINOVersion` as Pipeline is infered context.

szabi-luxonis commented:
> LGTM apart from the requested changes.
> Rationally behind the ImgDetections move from reference to getter?
> And in case it remains, add a setter as well to be able to create ImgDetections message and send it to the device

Switched back to reference, fixed setNumFramePools for detection NN, fixed exit on `q` press (waitkey) in all examples.
